File: src/main/java/io/github/boogiemonster1o1/usefulutilities/screen/ExpenseScreen.java
package io.github.boogiemonster1o1.usefulutilities.screen;

import io.github.boogiemonster1o1.usefulutilities.description.ExpenseDescription;
import io.github.boogiemonster1o1.usefulutilities.description.UtilitiesListDescription;
import io.github.cottonmc.cotton.gui.client.CottonClientScreen;

import java.io.IOException;

import net.minecraft.text.TranslatableText;

import static net.minecraft.client.MinecraftClient.getInstance;

public class ExpenseScreen extends CottonClientScreen {
    public ExpenseScreen() throws IOException {
        super(new TranslatableText("gui.utilities.expense"), new ExpenseDescription());
    }

    @Override
    public void onClose() {
        getInstance().openScreen(new CottonClientScreen(new UtilitiesListDescription()));
    }
}

File: src/main/scala/io/github/boogiemonster1o1/usefulutilities/description/screen/ExpenseScreen.scala
package io.github.boogiemonster1o1.usefulutilities.description.screen

import io.github.boogiemonster1o1.usefulutilities.description.{ExpenseDescription, UtilitiesListDescription}
import io.github.cottonmc.cotton.gui.client.CottonClientScreen
import net.minecraft.client.MinecraftClient
import net.minecraft.text.TranslatableText

import java.io.IOException

class ExpenseScreen @throws[IOException] extends CottonClientScreen(new TranslatableText("gui.utilities.expense"), new ExpenseDescription) {
	override def onClose(): Unit = {
		MinecraftClient.getInstance.setScreen(new CottonClientScreen(new UtilitiesListDescription))
	}
}
